### ISSUE TITLE
Generated code is not consistent across multiple runs of code generation

### ISSUE THREAD

**Issue by karthikraobr** [99designs/gqlgen#769]
### What happened?
The generated code changes over multiple runs of code generation. For the same schema file and setting.yml file, the below code kept changing between different runs of the code generation using gqlgen.
	
-			ret[i] = ec.marshalOCampaign2ᚖgithubᚗcomᚋorganizationᚋprojectᚋtype(ctx, sel, v[i])

+			ret[i] = ec.marshalNCampaign2ᚖgithubᚗcomᚋorganizationᚋprojectᚋtype(ctx, sel, v[i])

I did some poking around and found out that the issue was being caused by two types 

`type A{
   collection:[type!]!
}`

`type B{
   collection:[type]!
}`

As soon as I changed the above two types to 

`type A{
   collection:[type]!
}`

`type B{
   collection:[type]!
}`

the code being generated was consistent.
### What did you expect?
The generated code to be consistent across multiple runs.
### Minimal graphql.schema and models to reproduce
https://github.com/karthikraobr/gqlgen/tree/master/example/todo
Please check the git history of this repo and you will notice that although the schema and the settings file were untouched, the generated code changed.
Can be reproduced by repeatedly running gqlgen inside the todo directory.
### versions
 - `gqlgen version`? 
v0.9.1
 - `go version`?
1.12
 - dep or go modules?
modules

**Comment by karthikraobr** [comment]
The below methods are where the above line of code which keeps changing. Also, I have bound GraphQL types to our Protobuf messages.
```
func (ec *executionContext) marshalNCampaign2ᚕᚖgithubᚗcomᚋorganizationᚋprojectᚋtype(ctx context.Context, sel ast.SelectionSet, v []*type) graphql.Marshaler {
	ret := make(graphql.Array, len(v))
	var wg sync.WaitGroup
	isLen1 := len(v) == 1
	if !isLen1 {
		wg.Add(len(v))
	}
	for i := range v {
		i := i
		rctx := &graphql.ResolverContext{
			Index:  &i,
			Result: &v[i],
		}
		ctx := graphql.WithResolverContext(ctx, rctx)
		f := func(i int) {
			defer func() {
				if r := recover(); r != nil {
					ec.Error(ctx, ec.Recover(ctx, r))
					ret = nil
				}
			}()
			if !isLen1 {
				defer wg.Done()
			}
			ret[i] = ec.marshalOCampaign2ᚖgithubᚗcomᚋorganizationᚋprojectᚋtype(ctx, sel, v[i])
		}
		if isLen1 {
			f(i)
		} else {
			go f(i)
		}

	}
	wg.Wait()
	return ret
}
```

```
func (ec *executionContext) marshalNCampaign2ᚕᚖgithubᚗcomᚋorganizationᚋprojectᚋtype(ctx context.Context, sel ast.SelectionSet, v []*type) graphql.Marshaler {
	ret := make(graphql.Array, len(v))
	var wg sync.WaitGroup
	isLen1 := len(v) == 1
	if !isLen1 {
		wg.Add(len(v))
	}
	for i := range v {
		i := i
		rctx := &graphql.ResolverContext{
			Index:  &i,
			Result: &v[i],
		}
		ctx := graphql.WithResolverContext(ctx, rctx)
		f := func(i int) {
			defer func() {
				if r := recover(); r != nil {
					ec.Error(ctx, ec.Recover(ctx, r))
					ret = nil
				}
			}()
			if !isLen1 {
				defer wg.Done()
			}
			ret[i] = ec.marshalNCampaign2ᚖgithubᚗcomᚋorganizationᚋprojectᚋtype(ctx, sel, v[i])
		}
		if isLen1 {
			f(i)
		} else {
			go f(i)
		}

	}
	wg.Wait()
	return ret
}
```

**Comment by kfajardo** [comment]
Just updated gqlgen to be able to use this: https://github.com/99designs/gqlgen/blob/master/docs/content/reference/file-upload.md

Having a similar issue trying to add the scalar Upload.
Generated code basically looks like this:
<img width="949" alt="Screen Shot 2019-07-23 at 4 30 08 PM" src="https://user-images.githubusercontent.com/2063002/61696810-933d8f80-ad68-11e9-92b7-0ef282952ac8.png">

**Comment by vektah** [comment]
This is fixed in 0.10.2
